### PR TITLE
Fix issue where getting an unsuccessful response would delete all XForms

### LINKED ISSUE #376: Passing an incorrect username to "sync_deleted_projects" wipes out our database
When the ["sync_deleted_projects"](https://github.com/onaio/kaznet-web/blob/master/kaznet/apps/ona/api.py#L748) task either gets an invalid username or an unsuccessful request we end up deleting all our XForms.

### PR DESCRIPTION
**Pull request by DavisRayM**
Closes #376

### CODE REVIEW

**Comment by moshthepitt** [comment]
@DavisRayM Looks like we are assuming that if we ever receive an empty list of projects from Ona then something is wrong.  Is this a valid assumption?

Is there a way to actually check the response that we get from Ona i.e. is its not a 200 then abort

**Comment by moshthepitt** [comment]
@DavisRayM also, do we need to do this for forms and submissions?

**Comment by DavisRayM** [comment]
> @DavisRayM Looks like we are assuming that if we ever receive an empty list of projects from Ona then something is wrong. Is this a valid assumption?
> 
> Is there a way to actually check the response that we get from Ona i.e. is its not a 200 then abort

That was my assumption at the time, probably not the best assumption.

Currently we have no way to check that `status_code` for a request in most of the API functions since our `request` function, [here](https://github.com/onaio/kaznet-web/blob/master/kaznet/apps/ona/api.py#L120). We could check and make sure it's not a `None` and continue the process.

File: kaznet/apps/ona/api.py [hunk 1]
```diff
@@ -751,18 +751,21 @@ def sync_deleted_projects(usernames: list):
     If it finds any, it deletes them locally
     """
     onadata_project_pks = []
+
     for username in usernames:
         onadata_projects = get_projects(username=username)
+
         if isinstance(onadata_projects, list) and onadata_projects:
```
Comment (moshthepitt) [inline kaznet/apps/ona/api.py]: @DavisRayM it looks like if `onadata_projects` is `None` here then we should abort.

**Comment by DavisRayM** [comment]
> really nervous and curious about existing tests not needing an update @DavisRayM

Tests are currently failing, didn't try them out locally. I'll fix them

**Comment by DavisRayM** [comment]
@moshthepitt Fixed the failing tests but the tests that broke only broke due to a weird thing I was doing.... not sure if I should add more tests for this or not.. Any suggestion on what else should be tested within this work flow ?

**Comment by moshthepitt** [comment]
@DavisRayM do we need to do this for `sync_deleted_instances` as well?

**Comment by DavisRayM** [comment]
> @DavisRayM do we need to do this for `sync_deleted_instances` as well?

We currently don't need to make any code change to it. But we do need a test to make sure it never happens.

**Comment by DavisRayM** [comment]
> > @DavisRayM do we need to do this for `sync_deleted_instances` as well?
> 
> We currently don't need to make any code change to it. But we do need a test to make sure it never happens.

Added a test for this

**Pull request merged**